app/notification.go

```diff
@@ -15,13 +15,138 @@ import (
 	"github.com/mattermost/mattermost-server/store"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/mattermost/mattermost-server/utils/markdown"
+	"github.com/mitchellh/mapstructure"
 )
 
 const (
 	THREAD_ANY  = "any"
 	THREAD_ROOT = "root"
 )
 
+// TODO : Comment
+func (a *App) SendEphemeralPostWrapper(userId string, postDetails interface{}, post model.Post) {
+	err := mapstructure.Decode(postDetails, &post)
+	if err != nil {
+		panic(nil)
+	}
+	a.SendEphemeralPost(userId, &post)
+}
+
+// AddMentionedUsers will add the mentioned user id in the struct's list for mentioned users
+func (e *ExplicitMentions) AddMentionedUsers(ids []string) {
+	for _, id := range ids {
+		(*e).MentionedUserIds[id] = true
+	}
+}
+
+// CheckForMention checks if there is a mention to a specific user or to the keywords here / channel / all
+func (e *ExplicitMentions) CheckForMention(word string, keywords map[string][]string) bool {
```

Comment:
I think we can make all the methods private in this structure because is not used outside the `app` layer (actually is not used out of the `app/notification.go` file (I think).

Reply:
So i did a bit of sleuthing and it turns out that `GetChannelName`, `GetSenderName` and `SendNotifications` are also used outside `app/notification.go`. I will make the rest of the methods private ;)